emeritus-tech/landing-pages-libraries
# [CPT-431] Wharton Business Analytics Page Styling

🎟 [**Jira Ticket**](https://emeritus.atlassian.net/browseCPT-431)



### 🚢 Deployment Instructions

* N/A

### ✅ QA instructions

Smoke test the whole `dist/WBA.html` page taking into account the following:

* All modals should work as expected (without the `modals.js` file will only function in fallback mode, meaning that only clicking the `X` button will close it and `ESC` key or clicking outside will not)
* Testimonial slider should work as expected using `CSS` only (not entirely the same as the original landing page one)
* Responsiveness should work as expected

* To test Columbia WBA *Legacy* state:
  * Make sure body only has `.landing-page` class (it will have `.landing-page--course-preview` also by default, delete it for testing *Legacy* state)
  * Run command `npm run build` to compile.
  * Open `dist/WBA.html` on browser:
    * Design should look exactly as original WBA [landing page](https://app.zeplin.io/project/5ea8986a58a80025db39ddf7/screen/5ea765638c732ebeb69b0b59)
    * Custom iframe form container should be present on header (actual form is not being embebed)
    * All CTAs should take you to Hero custom form.

* To test Columbia WBA *Course Preview* state:
  * Make sure body has `.landing-page--course-preview` class (it will have it by default)
  * Run command `npm run build` to compile.
  * Open `dist/WBA.html` on browser:
    * Design should look exactly as course preview WBA [landing page](https://app.zeplin.io/project/5ea8986a58a80025db39ddf7/screen/5e963be12a1dc852e6c9930f)
    * Course Preview CTA should be present on header now
    * Course Preview Modal Form should be reachable trough CTAs (but actual iframe form is not present in modal)
    * Sticky nav should have now only the CTA instead of all the course information that *Legacy* state usually shows.

* To test Columbia WBA *Apply* state:
  * Make sure body has `.landing-page--apply` class (it will have `.landing-page--course-preview` also by default, delete it for testing *Apply* state)
  * Run command `npm run build` to compile.
  * Open `dist/WBA.html` on browser:
    * Design should look exactly as the new design for [WBA apply](https://app.zeplin.io/project/5ea8986a58a80025db39ddf7/screen/5ea7656a0f9de024096e2b1c)
    * Course Preview CTA should be present on header now
    * Course Preview Modal Form should be reachable trough CTAs (but actual iframe form is not present in modal)
    * Sticky nav will be the same of *Legacy* state.

* To test Columbia WBA *Thank You* state:
  * Make sure body has `.landing-page--thank-you` class (it will have `.landing-page--course-preview` also by default, delete it for testing *Thank You* state)
  * Run command `npm run build` to compile.
  * Open `dist/WBA.html` on browser:
    * Design should look exactly as the new design for [WBA thank you](https://app.zeplin.io/project/5ea8986a58a80025db39ddf7/screen/5ea76560c4cec8bf7b17e2e6)
    * Thank you message, phone number and apply button should now be present instead of form.
    * Sticky nav will be the same of *Legacy* state.


### ⚠️ Risks

* Low

### 🖼 Previews

Desktop (*Legacy* state)

![screencapture-file-Users-bocha-Code-rootstrap-landing-pages-libraries-styles-dist-WBA-html-2020-04-30-18_28_41](https://user-images.githubusercontent.com/251192/80761341-c50e2e80-8b10-11ea-9ec7-3c3fc0963da1.jpg)

* Hero (*Course Preview* state)

<img width="1680" alt="Screenshot 2020-04-30 17 47 59" src="https://user-images.githubusercontent.com/251192/80761109-54ffa880-8b10-11ea-9fdc-8a5a317a2be1.png">

* Hero (*Apply* state)

<img width="1680" alt="Screenshot 2020-04-30 16 17 29" src="https://user-images.githubusercontent.com/251192/80761091-4addaa00-8b10-11ea-9c3c-daca8fd5e58b.png">

* Hero (*Thank-you* state)

<img width="1679" alt="Screenshot 2020-04-30 17 45 51" src="https://user-images.githubusercontent.com/251192/80761067-41ecd880-8b10-11ea-8f29-65a01ca701fa.png">


### 📝 Notes

* Already implemented Kellogg and Columbia template also present new markup and styles for state variants.
* As requested, hero form and modal form wasn't implemented for now.
* Sticky nav script is present in `index.html` for testing purposes. Can be deleted after approval.
* Modals will not work entirely as expected unless original `modals.js` file is implemented.
